frontend/src/components/common/Header/Header.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import classNames from 'classnames';
+import { FaSearch } from 'react-icons/fa';
+import { FiShoppingBag } from 'react-icons/fi';
+import { NavLink } from 'react-router-dom';
+import { updateName, updateUrl, resetAll } from '../../../pages/Home/components/Criteria/filterSlice';
+import './Header.scss';
+
+const Header: React.FC = () => {
+  const navLinks = [
+    {
+      title: 'All',
+      slug: '/search/all',
+    },
+    {
+      title: 'New Arrival',
+      slug: '/search/new-arrivals',
+    },
+    {
+      title: 'Featured',
+      slug: '/search/featured',
+    },
+  ];
+  const [search, setSearch] = useState('');
+
+  const dispatch = useDispatch();
+  const handleSearch = () => {
+    dispatch(updateName(search));
+    dispatch(updateUrl());
+  };
+  const handleNavBar = (title: string) => {
+    if (title === 'All') {
+      dispatch(resetAll());
+    }
+  };
+  return (
+    <div className="header">
+      <div className="header__container">
+        <div className="header-navigation">
+          <div className="header-navigation__logo">
+            <img className="header-navigation__logo-img logo__icon" src="/images/vercel-icon.jpg" alt="Vercel Logo" />
+          </div>
+          <div className="header-navigation__links">
+            {navLinks.map(({ title, slug }) => {
+              return (
+                <NavLink
+                  className={({ isActive }) =>
+                    classNames('header-navigation__links-item', {"header-navigation__links-item--active": isActive})
+                  }
+                  key={title}
+                  to={slug}
+                  onClick={() => handleNavBar(title)}
+                >
+                  {title}
+                </NavLink>
+              );
+            })}
+          </div>
+        </div>
+
+        <div className="header-search">
+          <div className="header-search__container">
+            <input
+              className="header-search__input"
+              type="text"
+              placeholder="Search for products..."
+              onChange={(e) => setSearch(e.target.value)}
+              onKeyPress={(e) => {
```

Comment:
this type of behavior will be reused a lot, we can create a function to handle this. make it work for `enter` and `space`